scripts/updateapilist.py

```diff
@@ -137,37 +168,22 @@ def generate_table_contents(clients: List[CloudClient]) -> List[str]:
     return content_rows + pypi_links
 
 
-def allowed_repo(repo) -> bool:
+def allowed_split_repo(repo) -> bool:
     return (
         repo[REPO_RESPONSE_KEY].startswith("googleapis/python-")
         and repo[REPO_RESPONSE_KEY] not in REPO_EXCLUSION
         and not repo[REPO_RESPONSE_KEY]
```

Comment:
Observation: If you made this function allow the monorepo (and thus change the name back to `allowed_repo`, then the `Extractor` class would be simpler: it wouldn't need an `is_split_repo` field, and the list comprehension in `get_clients_from_batch_response` would be simpler.

The flip side is an extra string comparison for each repo at run-time. Which should still be fast, given everything else we're doing.

So, up to you....

I was thinking something like this

```suggestion
def allowed_split_repo(repo) -> bool:
    return (
        repo[REPO_RESPONSE_KEY].startswith(MONO_REPO)
        or (repo[REPO_RESPONSE_KEY].startswith("googleapis/python-")
             and repo[REPO_RESPONSE_KEY] not in REPO_EXCLUSION
             and not repo[REPO_RESPONSE_KEY])
```

Reply:
`REPO_RESPONSE_KEY` is specific to a split repo. For this to work, 
1. We would need to pass in the response key as an argument.

2. The value of repo[response_key] will evaluate to a package name which doesn't start with `googleapis/google-cloud-python`. Most package names  (but not all) are of the format `google-`.

We could do something like:

```python3

def allowed_repo(repo, response_key) -> bool:
    return repo[response_key].startswith("google-") or (
        repo[response_key].startswith("googleapis/python-")
        and repo[response_key] not in REPO_EXCLUSION
        and not repo[ARCHIVED_RESPONSE_KEY]
    )



```


But then again we'll be ignoring packages which don't have `google-` as a prefix. So we'll need to decide what we're checking for in a package.